plugins/fontdownloader.koplugin/main.lua

```diff
@@ -0,0 +1,519 @@
+    },
+--]]
+    recommended = {
+        "Bitter",
+        "Crimson Text",
+        "Gentium Book Basic",
+        "Ibarra Real Nova",
+        "Lit3rata",
+        "Merriweather",
+        "Source Serif Pro",
+    },
+    categories = {
+        ["display"]            = _("Display"),
+        ["handwriting"]        = _("Handwriting"),
+        ["monospace"]          = _("Monospace"),
+        ["sans-serif"]         = _("Sans serif"),
+        ["serif"]              = _("Serif"),
+    },
+    dates = {
+        ["last_week"]          = _("Last week"),
+        ["last_month" ]        = _("Last month"),
+        ["last_year" ]         = _("Last year"),
+        ["older"]              = _("Older"),
+    },
+    languages = {
+        ["arabic"]             = _("Arabic"),
+        ["bengali"]            = _("Bengali"),
+        ["chinese-simplified"] = _("Simplified Chinese"),
+        ["cyrillic"]           = _("Cyrillic"),
+        ["cyrillic-ext"]       = _("Cyrillic (extended)"),
+        ["devanagari"]         = _("Devanagari"),
+        ["greek"]              = _("Greek"),
+        ["greek-ext"]          = _("Greek (extended)"),
+        ["gujarati"]           = _("Gujarati"),
+        ["gurmukhi"]           = _("Gurmukhi"),
+        ["hebrew"]             = _("Hebrew"),
+        ["japanese"]           = _("Japanese"),
+        ["kannada"]            = _("Kannada"),
+        ["khmer"]              = _("Khmer"),
+        ["korean"]             = _("Korean"),
+        ["latin"]              = _("Latin"),
+        ["latin-ext"]          = _("Latin (extended)"),
+        ["malayalam"]          = _("Malayalam"),
+        ["myanmar"]            = _("Myanmar"),
+        ["oriya"]              = _("Oriya"),
+        ["sinhala"]            = _("Sinhala"),
+        ["tamil"]              = _("Tamil"),
+        ["telugu"]             = _("Telugu"),
+        ["thai"]               = _("Thai"),
+        ["tibetan"]            = _("Tibetan"),
+        ["vietnamese"]         = _("Vietnamese"),
+    },
+    variants = {
+        ["regular"]            = "regular",
+        ["italic"]             = "italic",
+        ["500"]                = "medium",
+        ["500italic"]          = "medium-italic",
+        ["700"]                = "bold",
+        ["700italic"]          = "bold-italic",
+    },
+}
+
+function FontDownloader:init()
+    self.ui.menu:registerToMainMenu(self)
+    local keyfile = io.open(self.user_key, "r")
+    if keyfile then
+        local key = keyfile:read("*a")
+        keyfile:close()
+        self.api_key = key
+        return
+    end
+    self.api_key = "AIzaSyDQZaihK8Lb7jJ3DYyrQrpyyJF7tyvrqAs"
+end
+
+function FontDownloader:addToMainMenu(menu_items)
+    -- plugin settings
+    menu_items.font_downloader = {
+        text = _("Set download location"),
+        sub_item_table = {
+            {
+                text = _("Set download location"),
```

Comment:
This goes directly into the More tools menu. I guess the first one should be `Font downloader`.
Also, it's a bit strange to have it that far away from the `Download more fonts`.

I'm also not sure that is really needed. I would download them in a subdirectory of our fonts/ directory, say `fonts/downloaded/` alongside `fonts/noto/` and `fonts/freefonts/` , so it's easier to clean after you have downloaded a bunch of crappy ones :)
(Unless on Android, the `fonts/` and `fonts/noto/` is not user-accessible ?)

Reply:
>  I would download them in a subdirectory of our fonts/ directory

getFontDir uses the platform external font directory, where users are expected to put fonts. I thought it would be handy to have a way to override that but probably we can avoid this option.